packages/cactus-test-plugin-htlc-eth-besu/src/test/typescript/integration/plugin-htlc-eth-besu/withdraw-endpoint.test.ts

```diff
@@ -0,0 +1,409 @@
+  const config = configService.newExampleConfigConvict(apiServerOptions);
+  const apiServer = new ApiServer({
+    httpServerApi: httpServer,
+    config: config.getProperties(),
+    pluginRegistry,
+  });
+  await apiServer.start();
+  t.comment("Start server");
+  test.onFinish(() => apiServer.shutdown());
+  const configuration = new Configuration({
+    basePath: node1Host,
+  });
+  t.comment("Setting configuration");
+
+  const api = new DefaultApi(configuration);
+  t.comment("Api generated");
+
+  //Deploy contract
+  const deployOut = await pluginHtlc.initialize(
+    keychainId,
+    accountRef,
+    privateKeyRef,
+    dataTest.estimated_gas,
+    Web3SigningCredentialType.PRIVATEKEYHEX,
+    connectorId,
+  );
+  t.comment(deployOut.transactionReceipt.contractAddress!);
+  t.ok(deployOut, "deployContract is truthy OK");
+  const hashTimeLockAddress = deployOut.transactionReceipt
+    .contractAddress as string;
+
+  //deploy DemoHelpers
+  const deployDemoHelper = await connector.deployContract({
+    web3SigningCredential: {
+      ethAccount: firstHighNetWorthAccount,
+      secret: besuKeyPair.privateKey,
+      type: Web3SigningCredentialType.PRIVATEKEYHEX,
+    },
+    bytecode: DemoHelpers.bytecode,
+    gas: dataTest.estimated_gas,
+  });
+  t.comment(deployDemoHelper.transactionReceipt.contractAddress!);
+  t.ok(deployDemoHelper, "deployContract DemoHelpers is OK");
+
+  const balance1 = await web3.eth.getBalance(dataTest.receiver);
+
+  //newContract
+  const bodyObj: NewContractObj = {
+    contractAddress: hashTimeLockAddress,
+    inputAmount: 10,
+    outputAmount: 0x04,
+    expiration: dataTest.expiration,
+    hashLock: dataTest.hashLock,
+    receiver: dataTest.receiver,
+    outputNetwork: "BTC",
+    outputAddress: "1AcVYm7M3kkJQH28FXAvyBFQzFRL6xPKu8",
+    keychainId: keychainId,
+    connectorId: connectorId,
+    accountRef: accountRef,
+    privateKeyRef: privateKeyRef,
+    credentialType: Web3SigningCredentialType.PRIVATEKEYHEX,
+    gas: dataTest.estimated_gas,
+  };
+  const resp = await api.newContract(bodyObj);
+  t.ok(resp, "new contract");
+
+  const { callOutput } = await connector.invokeContract({
+    contractAbi: DemoHelpers.abi,
+    contractAddress: deployDemoHelper.transactionReceipt
+      .contractAddress as string,
+    invocationType: EthContractInvocationType.CALL,
+    methodName: "getTxId",
+    params: [
+      firstHighNetWorthAccount,
+      dataTest.receiver,
+      10,
+      dataTest.hashLock,
+      dataTest.expiration,
+    ],
+    web3SigningCredential: {
+      ethAccount: firstHighNetWorthAccount,
+      type: Web3SigningCredentialType.PRIVATEKEYHEX,
+      secret: besuKeyPair.privateKey,
+    },
+    gas: dataTest.estimated_gas,
+  });
+  t.ok(callOutput, "result invoke is truthy OK");
+
+  const balance = await web3.eth.getBalance(dataTest.receiver);
+  t.equal(balance1, balance, "The balance of accout is OK");
+
+  // Test for invalid response test case
```

Comment:
@AzaharaC Code 500 should be returned for unknown errors. If the server validated the request and found that it had problems (such as something not being found that the request referred to as something in existence or something being in an invalid format) then a 400 or some other 4xx code could better indicate to end users that they need to fix something in their request.
In general 5xx codes mean that the backend is "at fault" meaning that the developer who sent the request has no action need, but in this case they do (which can be expressed in general with some 4xx codes that in general means that the backend thinks that the request sender is at fault)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

Reply:
https://devhints.io/http-status

Reply:
Done